service/test/BatchServerTest.cpp

```diff
@@ -426,3 +788,466 @@ TEST_F(BatchServerTest, fork_with_setup)
+        // because we are using the real POSIXSignal instead of the mock object.
+
+        // register the action_sigterm()
+        // This uses the real POSIXSignal instead of the mock object.
+        m_batch_server_test->child_register_handler();
+
+        int idx = 0;
+
+        for (const auto &request : m_signal_config) {
+        EXPECT_CALL(*m_pio_ptr, push_signal(request.name, request.domain,
+                                            request.domain_idx))
+            .WillOnce(Return(idx))
+            .RetiresOnSaturation();
+        ++idx;
+        }
+
+        for (const auto &request : m_control_config) {
+        EXPECT_CALL(*m_pio_ptr, push_control(request.name, request.domain,
+                                             request.domain_idx))
+            .WillOnce(Return(idx))
+            .RetiresOnSaturation();
+        ++idx;
+        }
+
+        // m_batch_status->receive_message() is called in BatchServerImp::read_message()
+        // Internally it is blocking on the read() call, waiting on a message from the client.
+        // Then the client sends it a SIGTERM, which activates action_sigterm()
+        // and receive_message() throws an exception in the check_return().
+        // This exception is caught in the BatchServerImp::read_message()
+        // and char in_message = BatchStatus::M_MESSAGE_TERMINATE is returned to BatchServerImp::event_loop()
+        EXPECT_CALL(*m_batch_status, receive_message())
+        .WillOnce(
+            [&write_pipe_fd](){
+                /* Extra code for synchronizing the server. */
+               char unique_char = '!';
+               int ret = write(write_pipe_fd, &unique_char, sizeof(unique_char));
+               ret = ret;
+
+                sleep(1024);
+                throw geopm::Exception(
+                    "BatchStatusImp: System call failed: " "read(2)",
+                    EINTR,
+                    __FILE__,
+                    __LINE__
+                );
+                // This is irrelevant, it doesn't get returned because of the throw statement.
+                // This is just to match the return type of BatchStatusImp::receive_message()
+                return '$';
+            }
+        )
+        .RetiresOnSaturation();
+
+        // The server process is stopped at BatchStatus::receive_message() inside BatchServerImp::read_message()
+        m_batch_server_test->run_batch();
+
+        // Allow Leak the mock objects in the child process.
+        testing::Mock::AllowLeak(m_pio_ptr.get());
+        testing::Mock::AllowLeak(m_batch_status.get());
+        testing::Mock::AllowLeak(m_signal_shmem.get());
+        testing::Mock::AllowLeak(m_control_shmem.get());
+    };
+
+    /// This function contains the parent process, which is the client.
+    /// It takes the read_pipe_fd and the PID of the child process, which is the server.
+    std::function<void(int, int)> parent_process_func = [this](int read_pipe_fd, int server_pid)
```

Comment:
```suggestion
    std::function<void(int, int)> parent_process_func = [](int read_pipe_fd, int server_pid)
```

Unless you're going to use `this`, it must be omitted since warnings are treated as errors.  It's an unused variable otherwise.